react-app/src/App.js

```diff
@@ -13,11 +13,10 @@ function App() {
     <Router>
       <div>
         <Switch>
-          <Route exact path="/get-started/:form" component={GetStartedPage} />
-          <Route exact path="/dashboard" component={Dashboard} />
-          <Route exact path="/dashboard/:clink" component={Dashboard} />
+          <Route exact path="/get-started/:form?" component={GetStartedPage} />
+          <Route exact path="/dashboard/:clinkId?/:searchParams?" component={Dashboard} />
```

Comment:
Do not use a subpath for search, use a query param instead
https://learnwithparam.com/blog/how-to-handle-query-params-in-react-router/